translator/util/ecsutil/ecsutil.go

```diff
@@ -48,13 +50,14 @@ func initECSUtilSingleton() (newInstance *ecsUtil) {
 	}
 	log.Println("I! attempt to access ECS task metadata to determine whether I'm running in ECS.")
 	ecsMetadataResponse, err := newInstance.getECSMetadata()
+
 	if err != nil {
 		log.Printf("I! access ECS task metadata fail with response %v, assuming I'm not running in ECS.\n", err)
 		return
 	}
 
 	newInstance.parseRegion(ecsMetadataResponse)
-	newInstance.Cluster = ecsMetadataResponse.Cluster
+	newInstance.parseClusterName(ecsMetadataResponse)
```

Comment:
Personally, I prefer explicitly setting the value like it did originally rather than expecting that the `parseClusterName` sets the value for the `Cluster` field, but I'm fine with this.

Reply:
I would prefer the same since its does not directly use the Cluster Name but the `Cluster ARN` in this case. Therefore, implicitly setting the value would be fine as long as its clarify the meaning though.